File: backend/app/schemas/platform_partner_dashboard.py
"""Platform partner dashboard schemas (read-only)."""

from __future__ import annotations

from pydantic import BaseModel, Field


class PlatformPartnerMetrics(BaseModel):
    total_revenue_inr: str
    platform_commission_inr: str
    active_customers: int
    active_laundries: int
    orders_total: int
    revenue_growth_pct: float | None = None
    orders_growth_pct: float | None = None


class PlatformPartnerChartPoint(BaseModel):
    date: str
    value: float
    label: str | None = None


class PlatformPartnerMonthPoint(BaseModel):
    month: str
    count: int


class PlatformPartnerCharts(BaseModel):
    revenue_trend: list[PlatformPartnerChartPoint] = Field(default_factory=list)
    orders_trend: list[PlatformPartnerChartPoint] = Field(default_factory=list)
    customer_growth: list[PlatformPartnerMonthPoint] = Field(default_factory=list)
    laundry_growth: list[PlatformPartnerMonthPoint] = Field(default_factory=list)


class PlatformPartnerTopLaundry(BaseModel):
    name: str
    city: str
    revenue_inr: str
    orders: int


class PlatformPartnerTopCity(BaseModel):
    city: str
    revenue_inr: str
    orders: int


class PlatformPartnerTopService(BaseModel):
    service_name: str
    revenue_inr: str
    quantity: int


class PlatformPartnerTables(BaseModel):
    top_laundries: list[PlatformPartnerTopLaundry] = Field(default_factory=list)
    top_cities: list[PlatformPartnerTopCity] = Field(default_factory=list)
    top_services: list[PlatformPartnerTopService] = Field(default_factory=list)


class PlatformPartnerDashboardResponse(BaseModel):
    metrics: PlatformPartnerMetrics
    charts: PlatformPartnerCharts
    tables: PlatformPartnerTables
    generated_at: str

File: frontend/services/platform-partner-dashboard.ts
import { api, type ApiEnvelope } from '@/lib/api';

export interface PlatformPartnerMetrics {
  total_revenue_inr: string;
  platform_commission_inr: string;
  active_customers: number;
  active_laundries: number;
  orders_total: number;
  revenue_growth_pct: number | null;
  orders_growth_pct: number | null;
}

export interface PlatformPartnerChartPoint {
  date: string;
  value: number;
  label: string | null;
}

export interface PlatformPartnerMonthPoint {
  month: string;
  count: number;
}

export interface PlatformPartnerCharts {
  revenue_trend: PlatformPartnerChartPoint[];
  orders_trend: PlatformPartnerChartPoint[];
  customer_growth: PlatformPartnerMonthPoint[];
  laundry_growth: PlatformPartnerMonthPoint[];
}

export interface PlatformPartnerTopLaundry {
  name: string;
  city: string;
  revenue_inr: string;
  orders: number;
}

export interface PlatformPartnerTopCity {
  city: string;
  revenue_inr: string;
  orders: number;
}

export interface PlatformPartnerTopService {
  service_name: string;
  revenue_inr: string;
  quantity: number;
}

export interface PlatformPartnerDashboard {
  metrics: PlatformPartnerMetrics;
  charts: PlatformPartnerCharts;
  tables: {
    top_laundries: PlatformPartnerTopLaundry[];
    top_cities: PlatformPartnerTopCity[];
    top_services: PlatformPartnerTopService[];
  };
  generated_at: string;
}

export async function getPlatformPartnerDashboard(): Promise<PlatformPartnerDashboard> {
  const { data } = await api.get<ApiEnvelope<PlatformPartnerDashboard>>('/platform-partner/dashboard');
  return data.data;
}

export function formatGrowthPct(pct: number | null | undefined): string {
  if (pct == null) return '—';
  return `${pct >= 0 ? '+' : ''}${pct}%`;
}
